faust-build/src/lib.rs

```diff
@@ -155,29 +153,27 @@ impl FaustBuilder {
 
         eprintln!("Wrote module:\n{}", dest_path.to_str().unwrap());
     }
-    pub fn build_xml(&self, out_dir: &str) {
-        eprintln!("cargo:rerun-if-changed={}", self.in_file);
+
+    pub fn build_xml_at_file(&self, out: &str) {
+        let gen_xml_fn = self.in_file.to_owned() + ".xml";
+        self.build_xml();
+        fs::rename(&gen_xml_fn, out).unwrap_or_else(|_| {
+            panic!(
+                "rename of xml file failed from '{:?}' to '{:?}'",
+                gen_xml_fn, out
+            )
+        });
+    }
+
+    pub fn build_xml(&self) {
         let mut output = Command::new(self.faust_path.clone().unwrap_or("faust".to_owned()));
 
-        let struct_name = match &self.struct_name {
-            Some(struct_name) => struct_name.clone(),
-            None => {
-                let dsp_path = PathBuf::from(&self.in_file);
-                dsp_path
-                    .file_stem()
-                    .unwrap()
-                    .to_string_lossy()
-                    .to_string()
-                    .to_camel_case()
-            }
-        };
+        let struct_name = self.get_struct_name();
 
         output
             .arg("-lang")
             .arg("rust")
             .arg("-xml")
-            .arg("--output-dir")
```

Comment:
Why did you remove this?

Reply:
i think it makes more sense to either rely on faust to do what it does by default. Or to specify a path with filename (like we do for the generated code) the output-dir is a third case right in the middle, so i would rather not provide this possible interface.